src/Shape.kt

```diff
@@ -0,0 +1,54 @@
+       /***** superclass called "Shape" that has two properties called shapeName and shapeColor
+           and also has two functions one called "info" and "computeArea"*****/
+
+open class Shape(val shapeName: String, val shapeColor: String) {
+    /*------------------------------------------------------------------------------*/
+        /***** functions called "info" that return name and color of shape.*****/
+    fun info() {
+        println("the shape name is:$shapeName and the color is:$shapeColor")
+    }
+    /*------------------------------------------------------------------------------*/
+        /***** functions called "computeArea" that has no parameter, and return Double data type.*****/
+    open fun computeArea(): Double {
+        return 0.0
+    }
+}
+    /*------------------------------------------------------------------------------*/
+        /***** subclass class called "Triangle" that return the area of Triangle.*****/
+class Triangle(var base: Double, var height: Double, shapeName: String, shapeColor: String) :
+    Shape(shapeName, shapeColor) {
+    override fun computeArea(): Double = (height * base) / 2
+}
+    /*------------------------------------------------------------------------------*/
+        /***** subclass class called "Rectangle" that return the area of Rectangle.*****/
+class Rectangle(var width: Double, var height: Double, shapeName: String, shapeColor: String) :
+    Shape(shapeName, shapeColor) {
+    override fun computeArea(): Double = height * width
+}
+    /*------------------------------------------------------------------------------*/
+        /***** subclass class called "Circle" that return the area of Circle.*****/
+class Circle(var radius: Double, shapeName: String, shapeColor: String) : Shape(shapeName, shapeColor) {
+    val pi: Double = 3.14
+    override fun computeArea(): Double = pi * (radius * radius)
```

Comment:
what dose override mean and what exactly is being changed ?

Reply:
we write "override" allow us to change the body of the functions we inherit from "computeArea()" and the functions we inherit from "computeArea()" most be open